internal/mode/static/state/graph/httproute_test.go

```diff
@@ -354,8 +354,18 @@ func TestBuildRoute(t *testing.T) {
 	hrInvalidFilters := createHTTPRoute("hr", gatewayNsName.Name, "example.com", "/filter")
 	addFilterToPath(hrInvalidFilters, "/filter", invalidFilter)
 
-	hrInvalidValidRules := createHTTPRoute("hr", gatewayNsName.Name, "example.com", invalidPath, "/filter", "/")
-	addFilterToPath(hrInvalidValidRules, "/filter", invalidFilter)
+	hrDroppedInvalidMatches := createHTTPRoute("hr", gatewayNsName.Name, "example.com", invalidPath, "/")
+
+	hrDroppedInvalidMatchesAndInvalidFilter := createHTTPRoute(
```

Comment:
nit: for consistency, since in other similar cases, it is plural:
```suggestion
	hrDroppedInvalidMatchesAndInvalidFilters := createHTTPRoute(
```